registry.go

```diff
@@ -48,6 +47,13 @@ func NewCodecMap() *CodecMap {
 // NetCodec is used to identify a network codec, that is, a network type for
 // which we are able to translate multiaddresses into standard Go net.Addr
 // and back.
+//
+// Deprecated: Unfortunately, these mappings aren't one to one. This abstraction
+// assumes that multiple "networks" can map to a single multiaddr protocol but
+// not the reverse. For example, this abstraction supports `tcp6, tcp4, tcp ->
+// /tcp/` really well but doesn't support `ip -> {/ip4/, /ip6/}`.
+//
+// Please use `RegisterFromNetAddr` and `RegisterToNetAddr` directly.
```

Comment:
^^ reason this abstraction doesn't work. Basically, it allows mapping multiple go "networks" to a single muliaddr protocol but not the inverse.